src/include/encoding.h

```diff
@@ -14,6 +14,22 @@
 #ifndef CEPH_ENCODING_H
 #define CEPH_ENCODING_H
 
+#include <set>
+#include <map>
+#include <deque>
+#include <vector>
+#include <string>
+#include <boost/optional/optional_io.hpp>
+#include <boost/tuple/tuple.hpp>
+
+#ifndef _BACKWARD_BACKWARD_WARNING_H
+#define _BACKWARD_BACKWARD_WARNING_H   // make gcc 4.3 shut up about hash_*
```

Comment:
before switching to gcc-7, we were using gcc 4.8 and up. guess it's a legacy of the C++98 era. can we take this opportunity to drop this `#define`? or make it a separated commit in this PR?